### PR TITLE
[Bot] Implement from issue #99: Create simple math utility with basic operations

### LINKED ISSUE #99: Create simple math utility with basic operations
﻿## Project Tag
project: math

## Description
Create a lightweight math utility module that provides basic arithmetic operations for use in other projects.

## Acceptance Criteria
- Function to add two numbers
- Function to multiply two numbers  
- Basic input validation
- Simple test to verify functions work
- Documentation with usage example

## Files to create
- \`math_utils.py\` - main utility functions
- \`test_math_utils.py\` - basic tests
- \`README.md\` - usage documentation

## Dependencies
- No external dependencies
- Use Python standard library only

## Scope & Segmentation Guidelines
**IMPORTANT FOR ANALYZER**: This is a simple, focused project. 
- Create maximum 3-4 logical tasks
- Group related activities together (don't create separate tasks for each function)
- Suggested segmentation: (1) Core implementation, (2) Testing, (3) Documentation
- Avoid micro-tasks like 'create __init__.py' or 'add single function'
- Keep tasks substantial and meaningful

This is a workflow verification test - prioritize logical task grouping over granular breakdown."

### PR DESCRIPTION
**Pull request by morotrade**
Auto-generated implementation.

Closes #99

Project root: `projects`

### CODE REVIEW

**Comment by morotrade** [comment]
### 🤖 AI Code Review



**Last Updated**: 2025-09-13 20:57:25 UTC

#### 📂 Project Root
`projects`

#### 📊 Summary
The implementation of the math utility is functional and includes basic tests. However, there are some organizational improvements and additional test cases needed for better coverage.

#### 🎯 Issue Counts
- **🚫 BLOCKER**: 0
- **⚠️ IMPORTANT**: 1  
- **💡 SUGGESTION**: 2

#### 🔍 Detailed Findings

#### ⚠️ IMPORTANT
- **`projects/math_utils.py:15`**: The main block for testing the utility functions is included in the same file as the implementation.
  *Suggestion*: Consider moving the main block to a separate script or a dedicated test file to improve code organization.

#### 💡 SUGGESTION
- **`projects/test_math_utils.py:9`**: The test for the 'add' function does not cover all edge cases.
  *Suggestion*: Add more test cases for the 'add' function, such as testing with negative numbers and zero.
- **`projects/test_math_utils.py:10`**: The test for the 'multiply' function is incomplete.
  *Suggestion*: Add tests for edge cases, such as multiplying by zero and negative numbers.


#### 🔧 Suggested Patches
The following unified diffs stay under the project root and can be applied by the developer:

**Patch 1**

```diff
--- a/projects/math_utils.py
+++ b/projects/math_utils.py
@@ -12,6 +12,8 @@
 
 if __name__ == "__main__":
     print("Math Utility")
     print("Add 2 + 3 =", add(2, 3))
     print("Multiply 2 * 3 =", multiply(2, 3))
+    # Consider moving this block to a separate test script or main execution file.
+
```

---
> 🔄 **Auto-Review Loop**: This comment updates automatically when you push changes to this branch.  
> 🏷️ **Labels**: `need-fix` = blockers to resolve, `ready-to-merge` = all clear!

**Comment by morotrade** [comment]
🔎 Using project root: `projects` for PR-fix
